### PR TITLE
Pagination fix

### LINKED ISSUE #141: Make Pagination return counts.
We need to return real query result counts. These counts can't be obtained easily using a second count query. Investigate other means to return these counts.

### PR DESCRIPTION
**Pull request by cacraig**
Fixes #141 . Makes pagination work, at a small cost to query performance. See #141 for discussion.

### CODE REVIEW

File: crime_data/common/base.py [hunk 1]
```diff
@@ -190,12 +190,12 @@ def _get(self, args):
 
         if args['output'] == 'csv':
             output = make_response(self.output_serialize(
-                self.with_metadata(qry, args), self.schema, 'csv', aggregate_many))
+                self.with_metadata(qry, args, over_count), self.schema, 'csv', aggregate_many))
```
Comment (harrisj) [inline crime_data/common/base.py]: Do you mind breaking this up over multiple lines?

File: crime_data/common/base.py [hunk 2]
```diff
@@ -299,15 +299,25 @@ def _stringify(self, data):
...
             (args['page'] - 1) * args['per_page'])
         if hasattr(paginated, 'data'):
             paginated = paginated.data
-        # count = results.count()
-        count = 10000
+
+        count = 0
+
+        # WINDOW FUNCTION COUNT(*) OVER () returns count in every row.
+        # Any row will give the correct count.
```
Comment (harrisj) [inline crime_data/common/base.py]: I think we have some whitespace issues here

File: crime_data/common/base.py [hunk 3]
```diff
@@ -299,15 +299,25 @@ def _stringify(self, data):
...
-        # count = results.count()
-        count = 10000
+
+        count = 0
+
+        # WINDOW FUNCTION COUNT(*) OVER () returns count in every row.
+        # Any row will give the correct count. 
+        try:
+            if over_count:
+                count = paginated[0][1] # Just select the first row.
```
Comment (harrisj) [inline crime_data/common/base.py]: Should we try to relax some of these style rules or stick to them?

File: crime_data/common/cdemodels.py [hunk 4]
```diff
@@ -345,6 +345,7 @@ def map(self):
 
     def base_query(self):
         """Gets root Query, based on class's base_table"""
+        #from sqlalchemy import func
```
Comment (harrisj) [inline crime_data/common/cdemodels.py]: If we don't need this anymore, can you delete this line?

File: crime_data/common/cdemodels.py [hunk 5]
```diff
@@ -651,6 +652,10 @@ class IncidentTableFamily(TableFamily):
                   models.NibrsLocationType.location_id), ))
     # TODO: COUNTY, TRIBE
 
+    def base_query(self):
+        """Gets root Query, based on class's base_table"""
+        from sqlalchemy import func
+        return db.session.query(self.base_table.table, func.count().over().label('count'))
```
Comment (harrisj) [inline crime_data/common/cdemodels.py]: Can you break this up to two separate lines?

File: crime_data/resources/incidents.py [hunk 6]
```diff
@@ -17,7 +17,8 @@ class IncidentsList(CdeResource):
     @use_args(marshmallow_schemas.ArgumentsSchema)
     @tuning_page
     def get(self, args):
-        return self._get(args)
+        over_count = True
```
Comment (harrisj) [inline crime_data/resources/incidents.py]: Is this `over_count` variable being used anywhere?

File: crime_data/common/base.py (hunk 2 above)
Comment (cacraig) [inline crime_data/common/base.py]: Not sure about this one. Seems okay to me?

**Comment by harrisj** [comment]
Thanks. LGTM. Merging.

**Pull request merged**